### ISSUE TITLE
GUI:  Completed targets reset after running the OTM

### ISSUE THREAD

**Issue by chazshapiro** [CaltechOpticalObservatories/NGPS#72]
Running the OTM un-completes completed targets in the main table

**Comment by astronomerdave** [comment]
One way to reproduce the issue:
```
$ seq targetset 98
$ seq test getnext
$ seq test completed
$ seq test getnext
$ seq test completed
```

This will load target set 98 and mark the first two targets as "completed". They will be grey in the GUI and written to the completed_obs database table.

Now click the "Accept" button in the GUI and it will clear the completed flag and set the target states to PENDING.

**Comment by chazshapiro** [comment]
This is not a small bug - I think the current design does not consider the logic of alternating actions by the OS and the OTM on the main table.  The OTM needs to know how to skip completed/inactive targets, return a plan for only the checked targets, and then the GUI must update the table and plot for only the checked targets (with click-highlights between the table and plot still intact).  There is also a possible race condition when in "Do All" mode.  We exit that mode when the user edits the table but what about when the OTM is run and is changing the "grey" table parts?

We need to think carefully how this should work, including how to "un-complete" targets intentionally so the list can be used again.  Maybe this could incorporate the checkboxes such that completed targets simply become unchecked and inactive, i.e. there is no separate **completed** status.

I'm adding the "missing feature" tag since, although this isn't explicitly documented, I think it is implicitly assumed that the GUI, OS, and OTM will coordinate in a sensible way.

**Comment by chazshapiro** [comment]
I stopped the OTM from overwriting the completed target state, but it still doesn't distinguish between targets when it runs, so the results are only valid when all targets are pending.

**Comment by chazshapiro** [comment]
I'm going to close this and re-open the remaining issues as a non-priority for commissioning

**Comment by jenniferwillow** [comment]
Hi Chaz,
     Would you like to discuss this issue at some point?
Sincerely,
Jennifer Milburn


On Oct 11, 2024, at 11:19 AM, Chaz Shapiro ***@***.******@***.***>> wrote:



Closed #72<https://github.com/CaltechOpticalObservatories/NGPS/issues/72> as completed.
